src/gluonts/torch/model/gnninf/estimator.py

```diff
@@ -0,0 +1,100 @@
+# Copyright 2018 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License").
+# You may not use this file except in compliance with the License.
+# A copy of the License is located at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# or in the "license" file accompanying this file. This file is distributed
+# on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+# express or implied. See the License for the specific language governing
+# permissions and limitations under the License.
+
+# If you use this code in your work please cite:
+# Multivariate Time Series Forecasting with Latent Graph Inference (https://arxiv.org/abs/2203.03423)
+
+
+import torch
+from gluonts.dataset.common import Dataset
+from gluonts.model.forecast_generator import DistributionForecastGenerator
+from gluonts.torch.model.predictor import PyTorchPredictor
+from . import trans
+from .module import GNNInfModule
+
+
+class GNNInfEstimator:
```

Comment:
I think it would make sense to inherit at least from the `Estimator` base class here.

There is also `PyTorchLightningEstimator` [here](https://github.com/awslabs/gluon-ts/blob/8f948c65c25ea888da4b6165fc3ec3fa487fb6b6/src/gluonts/torch/model/estimator.py#L38), but that requires you to set up a `LightningModule` class to wrap the model. The advantage is that you don't need to have your own training loop. But if you need to do something very custom about training, then it make sense to directly implement the `train` method your own way